sdks/go/pkg/beam/options/jobopts/options.go

```diff
@@ -169,3 +179,49 @@ func GetExperiments() []string {
 	}
 	return strings.Split(*Experiments, ",")
 }
+
+// GetPipelineResourceHints parses known standard hints and returns the flag set hints for the pipeline.
+// In case of duplicate hint URNs, the last value specified will be used.
+func GetPipelineResourceHints() resource.Hints {
+	hints := make([]resource.Hint, 0, len(ResourceHints))
+	for _, hint := range ResourceHints {
+		name, val, ok := strings.Cut(hint, "=")
+		if !ok {
+			panic(fmt.Sprintf("unparsable resource hint: %q", hint))
+		}
+		var h resource.Hint
+		switch name {
+		case "min_ram":
```

Comment:
Should we also check for equality to the fully specified payload (`beam:resources:min_ram_bytes:v1`)? Same thing with accelerator

Reply:
Done.

Reply:
To be clear here, it seems reasonable to maintain the correct semantics SDK side if we're able to, and it's not hard to add them to the switch statement. Also updated the test cases to ensure they're overriding as described in the flag.